components/logger.rst

```diff
@@ -21,16 +21,43 @@ Configuration variables:
 
 -  **baud_rate** (*Optional*, int): The baud rate to use for the serial
    UART port. Defaults to ``115200``. Set to ``0`` to disable logging via UART.
--  **tx_buffer_size** (*Optional*, string): The size of the buffer used
+-  **tx_buffer_size** (*Optional*, int): The size of the buffer used
    for log messages. Decrease this if you’re having memory problems.
-   Defaults to 512.
+   Defaults to ``512``.
+-  **hardware_uart** (*Optional*, string): The Hardware UART to use for logging.
+   Defaults to ``UART0``.
 -  **level** (*Optional*, string): The global log level. Any log message
-   with a lower severity will not be shown. Defaults to DEBUG.
+   with a lower severity will not be shown. Defaults to ``DEBUG``.
 -  **logs** (*Optional*, mapping): Manually set the log level for a
    specific component or tag. See :ref:`Manual Log Levels for more
    information <logger-manual_tag_specific_levels>`.
 -  **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
 
+.. _logger-hardware_uarts:
+
+Hardware UARTs
+--------------
+
+The logger component makes use of platform-specific hardware UARTs for serial logging.
+By default, the logger will occupy ``UART0``. The ESP32 has three hardware UARTs, all of
+which can be used for both transmit and receive. The ESP8266 only has two hardware UARTs,
+one of which is transmit-only. The ESP8266 ``UART0`` can also be 'swapped' to TX/RX on the
+CTS/RTS pins, if you need to use GPIO1 and GPIO3 for something else.
+
+Possible Hardware UART configurations:
+
+- ``UART0`` - TX: GPIO1, RX: GPIO3
+- ``UART0_SWAP`` - TX: GPIO15, RX: GPIO13  (Only on ESP8266)
+- ``UART1`` - TX: GPIO2, RX: None  (Only on ESP8266)
+- ``UART1`` - TX: GPIO9, RX: GPIO10  (Only on ESP832)
+- ``UART2`` - TX: GPIO16, RX: GPIO17  (Only on ESP832)
+
+.. note::
+
+   The ESP8266 Arduino framework has some logging that is hardcoded to use ``UART0`` by
+   default. If you want to send all messages via ``UART1`` on ESP8266, you will also need
+   to add ``build_flags: -DDEBUG_ESP_PORT=Serial1`` to your :ref:`esphomeyaml-platformio_options`.
```

Comment:
We can do better than this. In esphome[yaml] we have full control of the build flags (see https://github.com/esphome/esphome/pull/427/files#diff-ff8b4d8fa7ad3e06cc7e1dbdbc931debR101). We should set that build flag automatically.

(also because adding `build_flags:` to your platformio option will _replace_ all auto-generated ones, which will probably cause build issues)

Reply:
Oh, perfect. Will do.